build-macos-new.sh

```diff
@@ -28,8 +28,8 @@ brew install wget "$1" cmake || true
 
 CURRDIR=$(pwd)
 GTSAM_BRANCH="release/4.2"
-GTSAM_LIB_VERSION="4.2.0"
-GTSAM_PYTHON_VERSION="4.2.0"
+GTSAM_LIB_VERSION="4.2"
+GTSAM_PYTHON_VERSION="4.2"
```

Comment:
This is the Python version *not* the GTSAM version, so it should say 3.x, or better yet read the value from the CI as a passed parameter.